lib/puppet/util/windows/service.rb

```diff
@@ -741,16 +748,16 @@ def wait_on_state_transition(service, initial_state, final_state)
         state = nil
         elapsed_time = 0
         while elapsed_time <= DEFAULT_TIMEOUT
-          status = query_status(service)
-          state = status[:dwCurrentState]
 
+          query_status(service) do |status|
```

Comment:
Possible use-after-free for similar reasons per my comment in `transition_service_state`